pkg/gardenadm/resources.go

```diff
@@ -0,0 +1,114 @@
+	var (
+		cloudProfile *gardencorev1beta1.CloudProfile
+		project      *gardencorev1beta1.Project
+		shoot        *gardencorev1beta1.Shoot
+	)
+
+	if err := afero.Walk(f, configDir, func(path string, fileInfo fs.FileInfo, err error) error {
+		if err != nil {
+			return fmt.Errorf("failed walking directory %s: %w", configDir, err)
+		}
+
+		if fileInfo.IsDir() || !(strings.HasSuffix(path, ".yaml") || strings.HasSuffix(path, ".yml") || strings.HasSuffix(path, ".json")) {
+			return nil
+		}
+
+		content, err := f.ReadFile(path)
+		if err != nil {
+			return fmt.Errorf("failed reading file %s: %w", path, err)
+		}
+
+		var (
+			decoder    = yaml.NewYAMLOrJSONDecoder(bytes.NewReader(content), 1024)
+			decodedObj map[string]any
+		)
+
+		for indexInFile := 0; true; indexInFile++ {
+			objLog := log.WithValues("path", path, "indexInFile", indexInFile)
+
+			if err := decoder.Decode(&decodedObj); err == io.EOF {
+				break
+			} else if err != nil {
+				return fmt.Errorf("failed decoding resource at index %d in %s", indexInFile, path)
+			}
+
+			if decodedObj == nil {
+				continue
+			}
+
+			obj := &unstructured.Unstructured{Object: decodedObj}
+			objLog.V(2).Info("Read resource", "apiVersion", obj.GetAPIVersion(), "kind", obj.GetKind(), "name", obj.GetName(), "namespace", obj.GetNamespace())
+
+			switch gk := obj.GroupVersionKind().GroupKind(); gk {
+			case gardencorev1beta1.SchemeGroupVersion.WithKind("CloudProfile").GroupKind():
+				if cloudProfile != nil {
+					return fmt.Errorf("found more than one *gardencorev1beta1.CloudProfile resource, but only one is allowed")
+				}
+				cloudProfile = &gardencorev1beta1.CloudProfile{}
+				if err := kubernetes.GardenScheme.Convert(obj, cloudProfile, nil); err != nil {
+					return fmt.Errorf("failed converting object with group kind %s to CloudProfile: %w", gk, err)
+				}
+
+			case gardencorev1beta1.SchemeGroupVersion.WithKind("Project").GroupKind():
+				if project != nil {
+					return fmt.Errorf("found more than one *gardencorev1beta1.Project resource, but only one is allowed")
+				}
+				project = &gardencorev1beta1.Project{}
+				if err := kubernetes.GardenScheme.Convert(obj, project, nil); err != nil {
+					return fmt.Errorf("failed converting object with group kind %s to Project: %w", gk, err)
+				}
+
+			case gardencorev1beta1.SchemeGroupVersion.WithKind("Shoot").GroupKind():
+				if shoot != nil {
+					return fmt.Errorf("found more than one *gardencorev1beta1.Shoot resource, but only one is allowed")
+				}
+				shoot = &gardencorev1beta1.Shoot{}
+				if err := kubernetes.GardenScheme.Convert(obj, shoot, nil); err != nil {
+					return fmt.Errorf("failed converting object with group kind %s to Shoot: %w", gk, err)
+				}
+			}
+		}
+
+		return nil
+	}); err != nil {
+		return nil, nil, nil, err
+	}
+
+	if cloudProfile == nil {
+		return nil, nil, nil, fmt.Errorf("must provide a *gardencorev1beta1.CloudProfile resource but did not find any")
```

Comment:
```suggestion
		return nil, nil, nil, fmt.Errorf("must provide a *gardencorev1beta1.CloudProfile resource, but did not find any")
```

(same for project and shoot)